src/java.base/share/classes/jdk/internal/classfile/impl/ClassFileImpl.java

```diff
@@ -135,6 +141,35 @@ public ClassModel parse(byte[] bytes) {
         return new ClassImpl(bytes, this);
     }
 
+    @Override
+    public ClassModel parse(final MemorySegment bytes) {
+        AbstractMemorySegmentImpl amsi = (AbstractMemorySegmentImpl) bytes;
+        if (amsi.unsafeGetBase() instanceof byte[] ba) {
```

Comment:
there's also `MemorySegment::heapObject` if you don't want to drop down to unsafe. That method returns an empty optional if the segment is read-only -- but that might be good enough for a fast-path.